kernel/rtlil.cc

```diff
@@ -393,15 +399,24 @@ bool RTLIL::Const::as_bool() const
 
 int RTLIL::Const::as_int(bool is_signed) const
 {
-	bitvectorize_internal();
-	bitvectype& bv = get_bits();
 	int32_t ret = 0;
-	for (size_t i = 0; i < bv.size() && i < 32; i++)
+	if (is_str()) {
+		const std::string& s = get_str();
+		int size = GetSize(s);
+		for (int i = std::min(4, size); i > 0; i--)
```

Comment:
Would be worth adding some comments that we're doing safe truncation with std::min and sign extension below

Reply:
Done.